CR: ENGR00028177
Patch name: ENGR00028177_fix_mx27_vpu_clock_enable_disable.patch
Date: Thu Mar 8 17:46:49 2007 -0600
Source: Freescale Semiconductor, Inc.
Description: This patch fixes the clock enable/disable functions for MX27 VPU.
===================================================================
diff -uNpr linux-2.6.19.2.orig/arch/arm/mach-mx27/clock.c linux-2.6.19.2/arch/arm/mach-mx27/clock.c
--- linux-2.6.19.2.orig/arch/arm/mach-mx27/clock.c	2007-04-25 17:39:06.000000000 -0600
+++ linux-2.6.19.2/arch/arm/mach-mx27/clock.c	2007-04-25 17:39:09.000000000 -0600
@@ -193,14 +193,22 @@ static void _clk_fec_disable(struct clk 
 
 static int _clk_vpu_enable(struct clk *clk)
 {
-	_clk_pccr01_enable(CCM_PCCR1_VPU_BAUD_MASK, CCM_PCCR1_HCLK_VPU_MASK);
+	unsigned long reg;
+
+	reg = __raw_readl(CCM_PCCR1);
+	reg |= CCM_PCCR1_VPU_BAUD_MASK | CCM_PCCR1_HCLK_VPU_MASK;
+	__raw_writel(reg, CCM_PCCR1);
 
 	return 0;
 }
 
 static void _clk_vpu_disable(struct clk *clk)
 {
-	_clk_pccr01_disable(CCM_PCCR1_VPU_BAUD_MASK, CCM_PCCR1_HCLK_VPU_MASK);
+	unsigned long reg;
+
+	reg = __raw_readl(CCM_PCCR1);
+	reg &= ~(CCM_PCCR1_VPU_BAUD_MASK | CCM_PCCR1_HCLK_VPU_MASK);
+	__raw_writel(reg, CCM_PCCR1);
 }
 
 static int _clk_sahara2_enable(struct clk *clk)
